### ISSUE TITLE
[335] Eversong Ranger does not use melee attacks

### ISSUE THREAD

**Issue by MrSmite** [TrinityCore/TrinityCore#15602]
__Core:__ 747ea7c
__DB:__ TDB_full_335.59_2015_07_14 + updates

__Bug:__

The [Eversong Ranger](http://www.wowhead.com/npc=15938/eversong-ranger) does not use melee attacks when an enemy attacks it in melee range. It will draw its melee weapons and just stand there.

__Details:__

I was running by on a level 1 Priest and for fun I decided to aggro a couple [Plagueborne Pillager](http://www.wowhead.com/npc=15654/plaguebone-pillager). The Rangers managed to kill the first one with their bows but the second one broke through the ranks and began attacking a Ranger in the back of the group.

The Ranger under attack drew her weapon but did not fight back and the other rangers simply stood there with their weapons drawn, watching.

**Comment by tkrokli** [comment]
To me, it does not look very surprising when looking at their ability tab, both on http://wotlk.openwow.com/npc=15938#abilities and http://www.wowhead.com/npc=15938/eversong-ranger#abilities, they are only listed with the ranged shoot ability (http://www.wowhead.com/spell=6660/shoot). I don't know if this is the limiting factor or if they are supposed to be able to use melee fighting as well, or if they are supposed to be able to use "shoot" at melee range. Maybe a video from retail could shed more light on this?

I am slightly more miffed about the missing event where some of the http://www.wowhead.com/npc=15654/plaguebone-pillagers are supposed to run up in a group and attack the rangers periodically, regardless of if you have taken the quest [The Dead Scar](http://www.wowhead.com/quest=8475/the-dead-scar) or not. Unfortunately, I could only find one comment on wowhead (http://www.wowhead.com/quest=8475/the-dead-scar#comments:id=155399) mentioning this, from patch 2.1.3. The quest NPC [Ranger Jaela](http://www.wowhead.com/npc=15416/ranger-jaela)'s quest completion text says: ["We don't anticipate the Scourge letting up on their attacks any time soon, but we still appreciate any help we get guarding the Scar."](http://wow.gamepedia.com/Quest:The_Dead_Scar#Completion) , but I'm not sure if those 2 quotes are valid as proof for this event. This should probably be listed as a separate issue, if there is any interest at all in reporting such a cosmetic issue.

**Comment by MrSmite** [comment]
More observations:

*Skeleton A* = first Skeleton
*Skeleton B* = a second Skeleton
*Skeleton C* = a third Skeleton

1. Aggro *Skeleton A* and *Skeleton B* by proximity (eg: no attacks)
2. Lead them toward the Rangers, let's say that *Skeleton A* survives to melee range
3. Run down the hill and aggro *Skeleton C*
4. *Skeleton A* will immediately disengage and come after you, ignore it
5. Drop aggro from *Skeleton C* by running away
6. *Skeleton A* will immediately go back to attacking the Ranger

Note: At no time during this entire process will the Rangers ever shoot at *Skeleton A* once it has engaged in melee, even when it comes after you in step 4. If you kill *Skeleton A* the Rangers will reset and use ranged attacks at the next *Skeleton* you pull.

**Comment by Killyana** [comment]
This must be executed only once:
``` sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=15938 AND `source_type`=0 AND `id`<14;
UPDATE `smart_scripts` SET `id`= `id`-12 WHERE `entryorguid`=15938 AND `source_type`=0 AND `id`>13;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(15938,0,0,0,0,0,100,0,0,0,2200,3800,11,6660,64,0,0,0,0,2,0,0,0,0,0,0,0,"Eversong Ranger - In Combat CMC - Cast 'Shoot'"),
(15938,0,1,0,2,0,100,1,0,15,0,0,25,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Eversong Ranger - Between 0-15% Health - Flee For Assist (No Repeat)");
